jujugui/static/gui/src/app/components/deployment/add-credentials/add-credentials.js

```diff
@@ -32,22 +32,53 @@ YUI.add('deployment-add-credentials', function() {
     },
 
     /**
+      Validate the form fields.
+
+      @method _validate
+      @returns {Boolean} Whether the form is valid.
+    */
+    _validate: function() {
+      var refs = this.refs;
+      var fields = [
+        'templateAccessKey',
+        'templateName',
+        'templateRegion',
+        'templateSecretKey'
+      ];
+      var formValid = true;
+      fields.forEach(field => {
+        var valid = refs[field].validate();
```

Comment:
This should be OK, this can only break if someone removes that field from the component, in which case this code should also be updated.